063-topic-replication-factor-change.md

```diff
@@ -0,0 +1,271 @@
+Once accepted, the replication factor change is put into the Cruise Control's task queue and executed asynchronously.
+
+If partition's current replication factor is less than target replication factor, new replicas are added to the partition in two steps.
+Tentatively add new replicas in a rack-aware, round-robin way, and then further optimize location of new replicas with provided or default Goal list (same heuristic of the rebalance).
+Rack awareness property is guaranteed only when the rack information is in metadata, and target replication factor is less than number or racks.
+
+If the current replication factor of partition is larger than target replication factor, one or more follower replicas are removed from the partition.
+Replicas are removed following the reverse order of position in replica list of partition.
+
+The two required parameters for this endpoint are `topic` (regular expression to specify subject topics) and `replication_factor` (target replication factor).
+In order to batch multiple changes in the same request, the Topic Operator uses a JSON payload containing topics grouped by their replication factor.
+This is useful in case of bulk replicas change (e.g. GitOps updates), or when a backlog of replicas changes is created because of a long rebalance task.
+
+```sh
+$ curl -vv -X POST -H "Content-Type: application/json" -d '{replication_factor:{topic_by_replication_factor:{2:topic1|topic3|topic4,3:topic2|topic5}}}' \
+  "http://localhost:9090/kafkacruisecontrol/topic_configuration?skip_rack_awareness_check=true&dryrun=false&json=true" | jq
+> POST /kafkacruisecontrol/topic_configuration?skip_rack_awareness_check=true&dryrun=false&json=true HTTP/1.1
+> Host: localhost:9090
+> User-Agent: curl/8.0.1
+> Accept: */*
+> 
+} [91 bytes data]
+< HTTP/1.1 200 OK
+< Date: Thu, 07 Dec 2023 14:58:55 GMT
+< Set-Cookie: JSESSIONID=node0lfkx183ijq4pjkkjahsegfmp30.node0; Path=/
+< Expires: Thu, 01 Jan 1970 00:00:00 GMT
+< User-Task-ID: 5344ca89-351f-4565-8d0f-9aade00e053d
+< Content-Type: application/json;charset=utf-8
+< Cruise-Control-Version: 2.5.77-SNAPSHOT
+< Cruise-Control-Commit_Id: 535ea07e8847990cc02dde1f1be99b387dbeaf5b
+< Content-Length: 17051
+< Server: Jetty(9.4.52.v20230823)
+< 
+{ [17321 bytes data]
+{
+  "summary": {
+    "numIntraBrokerReplicaMovements": 0,
+    "numReplicaMovements": 125,
+    "onDemandBalancednessScoreAfter": 100,
+    "intraBrokerDataToMoveMB": 0,
+    "monitoredPartitionsPercentage": 100,
+    "provisionRecommendation": "",
+    "excludedBrokersForReplicaMove": [],
+    "excludedBrokersForLeadership": [],
+    "provisionStatus": "RIGHT_SIZED",
+    "onDemandBalancednessScoreBefore": 100,
+    "recentWindows": 5,
+    "dataToMoveMB": 0,
+    "excludedTopics": [],
+    "numLeaderMovements": 0
+  }
+  ...
+```
+
+The `skip_rack_awareness_check` parameter configures whether to skip the rack awareness sanity check or not (disabled by default).
```

Comment:
Just a nit ... The way I understand it, by default the `skip_rack_awareness_check` is set to false? Maybe this would make it more clear?

```suggestion
The `skip_rack_awareness_check` parameter configures whether to skip the rack awareness sanity check or not (defaults to `false`).
```

As it is a bit confusing if the rack awareness check is disabled by default or the _skipping_ of the rack awareness check is disabled by default.

Reply:
Agreed.